secure.go

```diff
@@ -232,3 +255,27 @@ func (s *Secure) Process(w http.ResponseWriter, r *http.Request) error {
 
 	return nil
 }
+
+// CSPNonce returns the nonce value associated with the present request. If no nonce has been generated it returns an empty string.
+func CSPNonce(c context.Context) string {
+	if val, ok := c.Value(nonceKey).(string); ok {
+		return val
+	}
+
+	return ""
+}
+
+type key int
+
+const nonceKey key = iota
+
+func randNonce() string {
+
+	var buf [cspNonceSize]byte
+	_, err := io.ReadFull(rand.Reader, buf[:])
+	if err != nil {
+		panic("entropy failed " + err.Error())
```

Comment:
I would really prefer not to `panic` here. Is there another option or another way to fail gracefully?

Reply:
Hmm, I'm not really sure if a better option is available here. We cannot return a predictable value, as that would allow bypassing CSP.

I suppose another option would be to bubble the error upwards and cause the request to fail with a 500. Feels a bit odd to have this middleware return a 500, but I suppose this "should never happen". Thoughts?

Reply:
I have no suggestions... this is a tough one. Let's leave it as a panic, but make a more descriptive message like `"CSP Nonce rand.Reader failed" + err.Error()`. After that we should update the readme to reflect that if you are using a dynamic nonce, it could potentially panic... but is highly unlikely.